### PR TITLE
[PS UI] FaviconAvatar component

### PR DESCRIPTION
**Pull request by jpcloureiro**
**Description**

FaviconAvatar component implementation, for the Permission System UI.

Default
image

With Error
image

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Screenshots/Recordings**

_If applicable, add screenshots or recordings to visualize the changes_

**Issue**

Progresses #4104

### CODE REVIEW

File: app/component-library/components/FaviconAvatar/FaviconAvatar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { foxImageUri } from './FaviconAvatar.data';
+
+describe('FaviconAvatar', () => {
+  it('should match the snapshot', () => {
+    const wrapper = shallow(
+      <FaviconAvatar size={BaseAvatarSize.Xl} imageUrl={foxImageUri} />,
+    );
+    expect(wrapper).toMatchSnapshot();
+  });
+});
```
Comment (georgewrmarshall) [inline app/component-library/components/FaviconAvatar/FaviconAvatar.test.tsx:14]: Could we add a test here for the error state?

File: app/component-library/components/FaviconAvatar/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# FaviconAvatar
+
+FaviconAvatar is an Image component that renders an icon that is provided in the form of an URL. This component is based on the [BaseAvatar](../BaseAvatar/BaseAvatar.tsx) component.
```
Comment (georgewrmarshall) [inline app/component-library/components/FaviconAvatar/README.md:3]: nit: grammer
```suggestion
FaviconAvatar is an Image component that renders an icon that is provided in the form of a URL. This component is based on the [BaseAvatar](../BaseAvatar/BaseAvatar.tsx) component.
```

**Review by georgewrmarshall — APPROVED** [review]
One tiny nit and one comment about adding a test for the error state otherwise smashed it out of the park 💯

- verified component code to the best of my ability
- verified unit tests (need to update snapshots otherwise working)
- verified all stories in storybook
- verified all knobs and knob options work in storybook
- verified markdown documentation

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by gantunesr — APPROVED** [review]
LGTM

**Pull request merged**